lib/logstash/filters/mutate.rb

```diff
@@ -268,7 +273,11 @@ def convert(event)
       else
         value = converter.call(original)
       end
-      event[field] = value
+      if value.nil?
```

Comment:
since convert_boolean returns the original value on failed conversion, is this conditional still needed?